### PR TITLE
Refactor big endian

### PR DESCRIPTION
**Pull request by saitama951**
I have refactored some of the code for mongo db connection, based on the comments
you can probably review it once

### CODE REVIEW

File: src/MongoDB.Bson/IO/ReadWriteLittleEndian.cs
Comment (Zangetsu112) [inline src/MongoDB.Bson/IO/ReadWriteLittleEndian.cs:1]: They haven't decided on a name for the file yet. I'll post a comment and lets see what they have to say.

File: src/MongoDB.Bson/IO/BsonStreamAdapter.cs [hunk 1]
```diff
@@ -497,8 +497,8 @@ public override void WriteDouble(double value)
         {
             ThrowIfDisposed();
             var bytes = new byte[8];
-            BinaryPrimitives.WriteInt64LittleEndian(bytes, BitConverter.DoubleToInt64Bits(value));
-            _stream.Write(bytes, 0, 8);
+            BinaryPrimitivesCompat.WriteDoubleLittleEndian(bytes, value);
+	    _stream.Write(bytes, 0, 8);
```
Comment (Zangetsu112) [inline src/MongoDB.Bson/IO/BsonStreamAdapter.cs]: Indent seems off @501

File: src/MongoDB.Bson/IO/ReadWriteLittleEndian.cs
Comment (saitama951) [reply src/MongoDB.Bson/IO/ReadWriteLittleEndian.cs:1]: sure :+1: . thanks

File: src/MongoDB.Driver.Core/Core/Connections/KeepAliveValues.cs [hunk 2]
```diff
@@ -33,7 +33,7 @@ public byte[] ToBytes()
             var bytes = new byte[12];
             BinaryPrimitives.WriteUInt32LittleEndian(new Span<byte>(bytes, 0, 4), OnOff);
             BinaryPrimitives.WriteUInt32LittleEndian(new Span<byte>(bytes, 0, 4), KeepAliveTime);
```
Comment (Zangetsu112) [inline src/MongoDB.Driver.Core/Core/Connections/KeepAliveValues.cs:35]: This doesn't look right, rather it should be something like:
`BinaryPrimitives.WriteUInt32LittleEndian(new Span<byte>(bytes, 4, 4), KeepAliveTime);`

File: src/MongoDB.Driver.Core/Core/Connections/KeepAliveValues.cs [hunk 3]
```diff
@@ -33,7 +33,7 @@ public byte[] ToBytes()
...
-            BinaryPrimitives.WriteUInt32LittleEndian(new Span<byte>(bytes, 0, 4), KeepAliveInterval);
+            BinaryPrimitives.WriteUInt32LittleEndian(new Span<byte>(bytes, 0, 8), KeepAliveInterval);
```
Comment (Zangetsu112) [inline src/MongoDB.Driver.Core/Core/Connections/KeepAliveValues.cs:36]: Similarly this should be:
`BinaryPrimitives.WriteUInt32LittleEndian(new Span<byte>(bytes, 8, 4), KeepAliveInterval);`

**Pull request closed without merging**